expression/column.go

```diff
@@ -157,6 +158,11 @@ func (col *Column) GetType() *types.FieldType {
 	return col.RetType
 }
 
+// GetTypeClass implements Expression interface.
+func (col *Column) GetTypeClass() types.TypeClass {
+	return col.TypeClass
```

Comment:
I intended to avoid check hybridType in [runtime](https://github.com/pingcap/tidb/blob/master/expression/builtin_cast.go#L263).

If an expression is hybrid type, we do not add a cast function above it,
so the TypeClass of it need to be changed to the target type which may not  
 be consistent with the field type.